testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/moduleslisting/DeploymentModulesListTestCase.java

```diff
@@ -0,0 +1,252 @@
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * List modules which are on deployment’s classpath
+ * /deployment=application_war_ear_name:list-modules(verbose=false|true)
+ * @author <a href="mailto:szhantem@redhat.com">Sultan Zhantemirov</a> (c) 2019 Red Hat, inc.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class DeploymentModulesListTestCase {
+
+    private static final String NODE_TYPE = "deployment";
+    private static final String INNER_JAR_ARCHIVE_NAME = "inner-jar-lib.jar";
+    private static final String EXAMPLE_MODULE_TO_EXCLUDE = "ibm.jdk";
+    private static final String INNER_WAR_ARCHIVE_NAME = "list-modules.war";
+    private static final String EAR_DEPLOYMENT_NAME = "list-modules-ear-test.ear";
+    private static final String USER_MODULE = "org.hibernate";
+    private static final String CUSTOM_SLOT = "5.0";
+
+    @ContainerResource
+    private static ManagementClient managementClient;
+
+    @Deployment
+    public static Archive<?> deploy() {
+
+        EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, EAR_DEPLOYMENT_NAME);
+        JavaArchive earLib = ShrinkWrap.create(JavaArchive.class);
+        earLib.addClass(DeploymentModulesListTestCase.class);
+        earLib.addAsManifestResource(EmptyAsset.INSTANCE, "emptyJarLibResource.properties");
+        earLib.addAsManifestResource(EmptyAsset.INSTANCE, "properties/nestedJarLib.properties");
+        ear.addAsLibraries(earLib);
+
+        WebArchive war = ShrinkWrap.create(WebArchive.class, INNER_WAR_ARCHIVE_NAME);
+        war.addClass(DeploymentModulesListTestCase.class);
+        war.add(EmptyAsset.INSTANCE, "META-INF/example.txt");
+        war.add(EmptyAsset.INSTANCE, "META-INF/properties/nested.properties");
+        war.addAsWebInfResource(DeploymentModulesListTestCase.class.getPackage(), "web.xml", "web.xml");
+
+        JavaArchive libJar = ShrinkWrap.create(JavaArchive.class, INNER_JAR_ARCHIVE_NAME);
+        libJar.addClass(DeploymentModulesListTestCase.class);
+
+        war.addAsLibraries(libJar);
+        ear.addAsModules(libJar, war);
+        ear.add(new StringAsset(prepareJBossEarDeploymentStructure()),"/META-INF/jboss-deployment-structure.xml");
+        ear.addAsResource(EmptyAsset.INSTANCE, "emptyEarResource");
+        ear.addAsManifestResource(DeploymentModulesListTestCase.class.getPackage(), "application.xml", "application.xml");
```

Comment:
Same here but this time it will complain about missing application.xml

Reply:
@yersan thanks a lot for the notes, it should be fixed now.